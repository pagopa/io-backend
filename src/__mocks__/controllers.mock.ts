import AuthenticationLockService from "../services/authenticationLockService";

import * as TE from "fp-ts/TaskEither";

// --------------------------------
// AuthenticationLockService Mock
// --------------------------------

export const isUserAuthenticationLockedMock = jest.fn(() => TE.of(false));

export const AuthenticationLockServiceMock = {
  isUserAuthenticationLocked: isUserAuthenticationLockedMock,
} as any as AuthenticationLockService;

// --------------------------------
// \ AuthenticationLockService Mock
// --------------------------------
