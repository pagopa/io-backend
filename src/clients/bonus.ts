import nodeFetch from "node-fetch";
import { Client, createClient } from "../../generated/io-bonus-api/client";

export function BonusAPIClient(
  token: string,
  baseUrl: string,
  fetchApi: typeof fetch = (nodeFetch as unknown) as typeof fetch
): Client<"ApiKey"> {
  return createClient<"ApiKey">({
    basePath: "",
    baseUrl,
    fetchApi,
    withDefaults: op => params =>
      op({
        ...params,
        // please refer to source api spec for actual header mapping
        // https://github.com/pagopa/io-functions-bonus/blob/master/openapi/index.yaml#:~:text=%20%20ApiKey:
        ApiKey: token
      })
  });
}

export type BonusAPIClient = typeof BonusAPIClient;
