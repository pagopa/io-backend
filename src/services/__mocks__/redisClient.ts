// tslint:disable:no-any

"use strict";

/**
 * {@inheritDoc}
 */
export default class RedisClient {
  public readonly map: any;

  /**
   * Class constructor.
   */
  constructor() {
    this.map = new Map();
  }

  /**
   * {@inheritDoc}
   * @see https://redis.io/commands/set
   */
  public set(key: string, value: string): number {
    // return 0 if field already exists in the hash and the value was updated.
    if (this.map.get(key) !== undefined) {
      this.map.set(key, value);
      return 0;
    }
    // return 1 if field is a new field in the hash and value was set.
    this.map.set(key, value);
    return 1;
  }

  /**
   * {@inheritDoc}
   */
  public get(
    key: string,
    callback: (err: any, value: any) => void
  ): string | void {
    const err = undefined;

    return callback(err, this.map.get(key));
  }

  /**
   * {@inheritDoc}
   */
  public hdel(_: string, key: string): void {
    this.map.delete(key);
  }
}
