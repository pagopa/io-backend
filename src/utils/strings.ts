import * as t from "io-ts";
import { ulid } from "ulid";
import * as validator from "validator";

import { tag, Tagged } from "./types";

// a generator of identifiers
export type ObjectIdGenerator = () => NonEmptyString;

// tslint:disable-next-line:no-useless-cast
export const ulidGenerator: ObjectIdGenerator = () => ulid() as NonEmptyString;

/**
 * A non-empty string
 */

interface INonEmptyStringTag {
  readonly kind: "INonEmptyStringTag";
}

export const NonEmptyString = tag<INonEmptyStringTag>()(
  t.refinement(t.string, s => s.length > 0, "non empty string")
);

export type NonEmptyString = t.TypeOf<typeof NonEmptyString>;

/**
 * A string guaranteed to have a length within the range [L,H)
 */

interface IWithinRangeStringTag<L extends number, H extends number> {
  readonly lower: L;
  readonly higher: H;
  readonly kind: "IWithinRangeStringTag";
}

export const WithinRangeString = <
  L extends number,
  H extends number,
  T extends IWithinRangeStringTag<L, H>
>(
  l: L,
  h: H
  // tslint:disable-next-line:no-any
): Tagged<T, any, string> =>
  tag<T>()(
    t.refinement(
      t.string,
      s => s.length >= l && s.length < h,
      `string of length >= ${l} and < ${h}`
    )
  );

export type WithinRangeString<L extends number, H extends number> = string &
  IWithinRangeStringTag<L, H>;

/**
 * A string that matches a pattern.
 */

interface IPatternStringTag<P extends string> {
  readonly pattern: P;
  readonly kind: "IPatternStringTag";
}

export const PatternString = <P extends string, T extends IPatternStringTag<P>>(
  p: P
  // tslint:disable-next-line:no-any
): Tagged<T, any, string> =>
  tag<T>()(
    t.refinement(
      t.string,
      s => s.match(p) !== null,
      `string that matches the pattern "${p}"`
    )
  );

export type PatternString<P extends string> = string & IPatternStringTag<P>;

/**
 * A string that represents a valid email address.
 */

interface IEmailStringTag {
  readonly kind: "IEmailStringTag";
}

export const EmailString = tag<IEmailStringTag>()(
  t.refinement(
    t.string,
    s =>
      validator.isEmail(s, {
        allow_display_name: false,
        allow_utf8_local_part: false,
        require_tld: true
      }),
    "string that represents an email address"
  )
);

export type EmailString = t.TypeOf<typeof EmailString>;

/**
 * A string that represents an IP (v4 or v6).
 */

interface IIPStringTag {
  readonly kind: "IIPStringTag";
}

export const IPString = tag<IIPStringTag>()(
  t.refinement(t.string, validator.isIP, "string that represents an IP address")
);

export type IPString = t.TypeOf<typeof IPString>;
