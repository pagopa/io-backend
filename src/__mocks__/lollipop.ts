import { JwkPublicKey } from "@pagopa/ts-commons/lib/jwk";
import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
import { AssertionRefSha256 } from "../../generated/lollipop-api/AssertionRefSha256";
import { AssertionRefSha512 } from "../../generated/lollipop-api/AssertionRefSha512";
import * as jose from "jose";

export const anAssertionRef = "sha256-6LvipIvFuhyorHpUqK3HjySC5Y6gshXHFBhU9EJ4DoM=" as AssertionRefSha256;
export const anotherAssertionRef = "sha512-6LvipIvFuhyorHpUqK3HjySC5Y6gshXHFBhU9EJ4DoM=" as AssertionRefSha512;

export const aJwkPubKey: JwkPublicKey = {
  kty: "EC",
  crv: "secp256k1",
  x: "Q8K81dZcC4DdKl52iW7bT0ubXXm2amN835M_v5AgpSE",
  y: "lLsw82Q414zPWPluI5BmdKHK6XbFfinc8aRqbZCEv0A"
};

export const anEncodedJwkPubKey = jose.base64url.encode(
  JSON.stringify(aJwkPubKey)
) as NonEmptyString;

export const aLollipopAssertion = `<samlp:Response Destination="https://that.spid.example.org/saml2/acs/post" ID="_5e728601-9ad4-4686-b269-81d107a8194a" InResponseTo="${anotherAssertionRef}" IssueInstant="2021-02-04T15:41:59Z" Version="2.0" xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol">
    <saml:Issuer Format="urn:oasis:names:tc:SAML:2.0:nameid-format:entity">
        http://localhost:8080
    </saml:Issuer>
</samlp:Response>` as NonEmptyString;
