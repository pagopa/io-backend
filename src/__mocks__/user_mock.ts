import { FiscalCode, NonEmptyString } from "@pagopa/ts-commons/lib/strings";
import { EmailAddress } from "../../generated/backend/EmailAddress";
import { InitializedProfile } from "../../generated/backend/InitializedProfile";
import { SpidLevelEnum } from "../../generated/backend/SpidLevel";
import { IsInboxEnabled } from "@pagopa/io-functions-app-sdk/IsInboxEnabled";
import { IsWebhookEnabled } from "@pagopa/io-functions-app-sdk/IsWebhookEnabled";
import {
  PreferredLanguage,
  PreferredLanguageEnum
} from "@pagopa/io-functions-app-sdk/PreferredLanguage";
import { ServicePreferencesSettings } from "@pagopa/io-functions-app-sdk/ServicePreferencesSettings";
import { ServicesPreferencesModeEnum } from "@pagopa/io-functions-app-sdk/ServicesPreferencesMode";
import {
  BPDToken,
  FIMSToken,
  MyPortalToken,
  SessionToken,
  WalletToken,
  ZendeskToken
} from "../types/token";
import { User } from "../types/user";

export const aTimestamp = 1518010929530;
export const aFiscalCode = "GRBGPP87L04L741X" as FiscalCode;
export const anEmailAddress = "garibaldi@example.com" as EmailAddress;
export const aSpidEmailAddress = "garibaldi@spid.com" as EmailAddress;
export const aValidName = "Giuseppe Maria" as NonEmptyString;
export const aValidFamilyname = "Garibaldi" as NonEmptyString;
export const aValidDateofBirth = "2000-06-02";
export const aValidSpidLevel = SpidLevelEnum["https://www.spid.gov.it/SpidL2"];
export const aSessionTrackingId = "a-ssn-id";
export const mockSessionToken = "c77de47586c841adbd1a1caeb90dce25dcecebed620488a4f932a6280b10ee99a77b6c494a8a6e6884ccbeb6d3fe736b" as SessionToken;
export const mockWalletToken = "5ba5b99a982da1aa5eb4fd8643124474fa17ee3016c13c617ab79d2e7c8624bb80105c0c0cae9864e035a0d31a715043" as WalletToken;
export const mockMyPortalToken = "c4d6bc16ef30211fb3fa8855efecac21be04a7d032f8700dc4d6bc16ef30211fb3fa8855efecac21be04a7d032f8700d" as MyPortalToken;
export const mockBPDToken = "4123ee213b64955212ea59e3beeaad1e5fdb3a36d22104164123ee213b64955212ea59e3beeaad1e5fdb3a36d2210416" as BPDToken;
export const mockZendeskToken = "aaaa12213b64955212ea59e3beeaad1e5fdb3a36d2210abcaaaa12213b64955212ea59e3beeaad1e5fdb3a36d2210abc" as ZendeskToken;
export const mockFIMSToken = "aaaa12213b64955212ea59e3beeaad1e5fdb3a36d2210bcdaaaa12213b64955212ea59e3beeaad1e5fdb3a36d2210bcd" as FIMSToken;

// mock for a valid User
export const mockedUser: User = {
  created_at: aTimestamp,
  family_name: aValidFamilyname,
  fiscal_code: aFiscalCode,
  name: aValidName,
  date_of_birth: aValidDateofBirth,
  session_token: mockSessionToken,
  spid_email: aSpidEmailAddress,
  spid_level: aValidSpidLevel,
  wallet_token: mockWalletToken,
  myportal_token: mockMyPortalToken,
  bpd_token: mockBPDToken,
  zendesk_token: mockZendeskToken,
  fims_token: mockFIMSToken,
  session_tracking_id: aSessionTrackingId
};

export const aCustomEmailAddress = "custom-email@example.com" as EmailAddress;

export const anIsInboxEnabled = true as IsInboxEnabled;
export const anIsWebookEnabled = true as IsWebhookEnabled;
export const aPreferredLanguages: ReadonlyArray<PreferredLanguage> = [
  PreferredLanguageEnum.it_IT
];
export const aServicePreferencesSettings: ServicePreferencesSettings = {
  mode: ServicesPreferencesModeEnum.AUTO
};

export const mockedInitializedProfile: InitializedProfile = {
  email: aCustomEmailAddress,
  family_name: mockedUser.family_name,
  fiscal_code: mockedUser.fiscal_code,
  has_profile: true,
  is_email_enabled: true,
  is_email_validated: true,
  is_inbox_enabled: anIsInboxEnabled,
  is_webhook_enabled: anIsWebookEnabled,
  name: mockedUser.name,
  preferred_languages: aPreferredLanguages,
  service_preferences_settings: aServicePreferencesSettings,
  version: 42
};
