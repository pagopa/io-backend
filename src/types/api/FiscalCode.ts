/**
 * User's fiscal code.
 */

import { PatternString } from "../../utils/strings";

import * as t from "io-ts";

export type FiscalCode = t.TypeOf<typeof FiscalCode>;

export const FiscalCode = PatternString(
  "^[A-Z]{6}[0-9LMNPQRSTUV]{2}[ABCDEHLMPRST][0-9LMNPQRSTUV]{2}[A-Z][0-9LMNPQRSTUV]{3}[A-Z]$"
);
