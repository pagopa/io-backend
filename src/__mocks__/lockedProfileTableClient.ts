import { TableClient, TableInsertEntityHeaders } from "@azure/data-tables";

import { aFiscalCode } from "./user_mock";
import { FiscalCode } from "@pagopa/ts-commons/lib/strings";

export async function* profileLockedRecordIterator() {
  yield { partitionKey: aFiscalCode, rowKey: "123456789" };
}
export async function* noProfileLockedRecordIterator(): ReturnType<
  typeof profileLockedRecordIterator
> {}
export async function* errorProfileLockedRecordIterator(): ReturnType<
  typeof profileLockedRecordIterator
> {
  //Sonarcloud requires at least one `yield` before `throw` operation
  yield { partitionKey: "CF" as FiscalCode, rowKey: "" };
  throw new Error("an Error");
}

export const listLockedProfileEntitiesMock = jest.fn(
  noProfileLockedRecordIterator
);

export const createEntityMock = jest.fn(
  async () => ({} as TableInsertEntityHeaders)
);

export const lockedProfileTableClient: TableClient = {
  createEntity: createEntityMock,
  listEntities: listLockedProfileEntitiesMock,
} as unknown as TableClient;
