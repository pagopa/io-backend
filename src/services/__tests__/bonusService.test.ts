// tslint:disable: no-duplicate-string no-identical-functions

import * as t from "io-ts";
import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";

import { FiscalCode } from "../../../generated/backend/FiscalCode";

import { BonusActivation } from "../../../generated/io-bonus-api/BonusActivation";
import { BonusActivationStatusEnum } from "../../../generated/io-bonus-api/BonusActivationStatus";
import { BonusCode } from "../../../generated/io-bonus-api/BonusCode";
import { EligibilityCheck } from "../../../generated/io-bonus-api/EligibilityCheck";
import { PaginatedBonusActivationsCollection } from "../../../generated/io-bonus-api/PaginatedBonusActivationsCollection";
import { BonusAPIClient } from "../../clients/bonus";
import { mockedUser } from "../../__mocks__/user_mock";
import BonusService from "../bonusService";

const aBonusId = "aBonusId" as NonEmptyString & BonusCode;

const aEligibilityCheck: EligibilityCheck = {
  dsu_request: {
    dsu_created_at: new Date(),
    dsu_protocol_id: "123" as NonEmptyString,
    family_members: [],
    has_discrepancies: false,
    isee_type: "123" as NonEmptyString,
    max_amount: 150,
    max_tax_benefit: 30,
    request_id: 1
  },
  id: "aEligibilityCheck.id" as NonEmptyString,
  // tslint:disable-next-line: no-any
  status: "ELIGIBLE" as any
};

const aBonusActivation: BonusActivation = {
  applicant_fiscal_code: "SPNDNL80R14C522K" as FiscalCode,
  created_at: new Date(),
  dsu_request: {
    dsu_created_at: new Date(),
    dsu_protocol_id: "dsuprotid" as NonEmptyString,
    family_members: [
      {
        fiscal_code: "SPNDNL80R14C522K" as FiscalCode,
        name: "mario" as NonEmptyString,
        surname: "rossi" as NonEmptyString
      }
    ],
    has_discrepancies: false,
    isee_type: "iseetype",
    max_amount: 150,
    max_tax_benefit: 30,
    request_id: 1
  },
  id: aBonusId,
  status: BonusActivationStatusEnum.ACTIVE
};

const aPaginatedBonusActivationCollection: PaginatedBonusActivationsCollection = {
  items: [
    {
      id: aBonusId,
      is_applicant: true
    }
  ]
};

// mock for a not adult User

const mockGetAllBonusActivations = jest.fn();
const mockGetBonusEligibilityCheck = jest.fn();
const mockGetLatestBonusActivationById = jest.fn();
const mockStartBonusActivationProcedure = jest.fn();
const mockStartBonusEligibilityCheck = jest.fn();

const mockBonusAPIClient = {
  getAllBonusActivations: mockGetAllBonusActivations,
  getBonusEligibilityCheck: mockGetBonusEligibilityCheck,
  getLatestBonusActivationById: mockGetLatestBonusActivationById,
  startBonusActivationProcedure: mockStartBonusActivationProcedure,
  startBonusEligibilityCheck: mockStartBonusEligibilityCheck
} as ReturnType<BonusAPIClient>;

const api = mockBonusAPIClient;

describe("BonusService#getBonusEligibilityCheck", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should make the correct api call", async () => {
    const service = new BonusService(api);

    await service.getBonusEligibilityCheck(mockedUser);

    expect(mockGetBonusEligibilityCheck).toHaveBeenCalledWith({
      fiscalcode: mockedUser.fiscal_code
    });
  });

  it("should handle a successful request", async () => {
    mockGetBonusEligibilityCheck.mockImplementation(() =>
      t.success({ status: 200, value: aEligibilityCheck })
    );

    const service = new BonusService(api);

    const res = await service.getBonusEligibilityCheck(mockedUser);

    expect(res).toMatchObject({
      kind: "IResponseSuccessJson",
      value: aEligibilityCheck
    });
  });

  it("should handle an internal error response", async () => {
    const aGenericProblem = {};
    mockGetBonusEligibilityCheck.mockImplementation(() =>
      t.success({ status: 500, value: aGenericProblem })
    );

    const service = new BonusService(api);

    const res = await service.getBonusEligibilityCheck(mockedUser);

    expect(res).toMatchObject({
      kind: "IResponseErrorInternal"
    });
  });

  it("should return an error for unhandled response status code", async () => {
    mockGetBonusEligibilityCheck.mockImplementation(() =>
      t.success({ status: 123 })
    );
    const service = new BonusService(api);

    const res = await service.getBonusEligibilityCheck(mockedUser);

    expect(res).toMatchObject({
      kind: "IResponseErrorInternal"
    });
  });

  it("should return an error if the api call thows", async () => {
    mockGetBonusEligibilityCheck.mockImplementation(() => {
      throw new Error();
    });
    const service = new BonusService(api);

    const res = await service.getBonusEligibilityCheck(mockedUser);

    expect(res).toMatchObject({
      kind: "IResponseErrorInternal"
    });
  });
});

describe("BonusService#getLatestBonusActivationById", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should make the correct api call", async () => {
    const service = new BonusService(api);

    await service.getLatestBonusActivationById(mockedUser, aBonusId);

    expect(mockGetLatestBonusActivationById).toHaveBeenCalledWith({
      bonus_id: aBonusId,
      fiscalcode: mockedUser.fiscal_code
    });
  });

  it("should handle a successful request", async () => {
    mockGetLatestBonusActivationById.mockImplementation(() =>
      t.success({ status: 200, value: aBonusActivation })
    );

    const service = new BonusService(api);

    const res = await service.getLatestBonusActivationById(
      mockedUser,
      aBonusId
    );

    expect(res).toMatchObject({
      kind: "IResponseSuccessJson",
      value: {
        ...aBonusActivation,
        qr_code: [
          { content: expect.any(String), mime_type: "image/png" },
          { content: expect.any(String), mime_type: "image/svg+xml" }
        ]
      }
    });
    if (res.kind === "IResponseSuccessJson") {
      const svg = new Buffer(res.value.qr_code[1].content, "base64");
      expect(svg.toString()).toContain(`fill=\"black\"`);
    }
  });

  it("should handle an internal error response", async () => {
    const aGenericProblem = {};
    mockGetLatestBonusActivationById.mockImplementation(() =>
      t.success({ status: 500, value: aGenericProblem })
    );

    const service = new BonusService(api);

    const res = await service.getLatestBonusActivationById(
      mockedUser,
      aBonusId
    );

    expect(res).toMatchObject({
      kind: "IResponseErrorInternal"
    });
  });

  it("should return an error for unhandled response status code", async () => {
    mockGetLatestBonusActivationById.mockImplementation(() =>
      t.success({ status: 123 })
    );
    const service = new BonusService(api);

    const res = await service.getLatestBonusActivationById(
      mockedUser,
      aBonusId
    );

    expect(res).toMatchObject({
      kind: "IResponseErrorInternal"
    });
  });

  it("should return an error if the api call thows", async () => {
    mockGetLatestBonusActivationById.mockImplementation(() => {
      throw new Error();
    });
    const service = new BonusService(api);

    const res = await service.getLatestBonusActivationById(
      mockedUser,
      aBonusId
    );

    expect(res).toMatchObject({
      kind: "IResponseErrorInternal"
    });
  });
});

describe("BonusService#getAllBonusActivations", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should make the correct api call", async () => {
    const service = new BonusService(api);

    await service.getAllBonusActivations(mockedUser);

    expect(mockGetAllBonusActivations).toHaveBeenCalledWith({
      fiscalcode: mockedUser.fiscal_code
    });
  });

  it("should handle a successful request", async () => {
    mockGetAllBonusActivations.mockImplementation(() =>
      t.success({
        status: 200,
        value: aPaginatedBonusActivationCollection
      })
    );

    const service = new BonusService(api);

    const res = await service.getAllBonusActivations(mockedUser);

    expect(res).toMatchObject({
      kind: "IResponseSuccessJson"
    });
  });

  it("should handle no found bonus", async () => {
    mockGetAllBonusActivations.mockImplementation(() =>
      t.success({ status: 200, value: { items: [], page_size: 0 } })
    );

    const service = new BonusService(api);

    const res = await service.getAllBonusActivations(mockedUser);

    expect(res).toMatchObject({
      kind: "IResponseSuccessJson",
      value: { items: [], page_size: 0 }
    });
  });

  it("should handle an internal error response", async () => {
    const aGenericProblem = {};
    mockGetAllBonusActivations.mockImplementation(() =>
      t.success({ status: 500, value: aGenericProblem })
    );

    const service = new BonusService(api);

    const res = await service.getAllBonusActivations(mockedUser);

    expect(res).toMatchObject({
      kind: "IResponseErrorInternal"
    });
  });

  it("should return an error for unhandled response status code", async () => {
    mockGetAllBonusActivations.mockImplementation(() =>
      t.success({ status: 123 })
    );
    const service = new BonusService(api);

    const res = await service.getAllBonusActivations(mockedUser);

    expect(res).toMatchObject({
      kind: "IResponseErrorInternal"
    });
  });

  it("should return an error if the api call thows", async () => {
    mockGetAllBonusActivations.mockImplementation(() => {
      throw new Error();
    });
    const service = new BonusService(api);

    const res = await service.getAllBonusActivations(mockedUser);

    expect(res).toMatchObject({
      kind: "IResponseErrorInternal"
    });
  });
});
