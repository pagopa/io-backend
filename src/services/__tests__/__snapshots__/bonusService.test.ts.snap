// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BonusService#getLatestBonusActivationById should handle a successful request 1`] = `
Object {
  "apply": [Function],
  "kind": "IResponseSuccessJson",
  "value": Object {
    "applicant_fiscal_code": "SPNDNL80R14C522K",
    "created_at": 2020-06-18T10:15:11.704Z,
    "dsu_request": Object {
      "dsu_created_at": "",
      "dsu_protocol_id": "dsuprotid",
      "family_members": Array [
        Object {
          "fiscal_code": "SPNDNL80R14C522K",
          "name": "mario",
          "surname": "rossi",
        },
      ],
      "has_discrepancies": false,
      "isee_type": "iseetype",
      "max_amount": 150,
      "max_tax_benefit": 30,
      "request_id": 1,
    },
    "id": "aBonusId",
    "qr_code": Array [
      Object {
        "content": "77+9UE5HDQoaCgAAAA1JSERSAAAA77+9AAAA77+9CAAAAADvv70F77+9AAABHklEQVR477+977+977+9QRLvv70wCABA77+977+977+977+9Awbvv70kaDvvv71rLVkPVCDvv73vv73vv71tHURERERERERERETvv70pOu+/vVd8XSkKEVHvv71o77+9DCd777+9RCEi77+9Fw1z77+9JO+/ve+/vRAR77+9TXQR77+977+977+9R0Vn77+9Fl9HRO+/vQ5R77+977+9Ie+/ve+/ve+/ve+/ve+/vWhZT++/vc2w77+9Lu+/ve+/vWjvv70cMu+/vVPvv71k77+977+9KO+/vRQpJU3vv71m77+977+9QxIR77+977+977+9HkZI77+977+9456C77+977+9VRQX77+9Nyfvv73vv70REW0V77+9ae+/ve+/ve+/ve6BiO+/vUE01Zfvv70cORER77+9Eu+/ve+/ve+/vUQd77+9Le+/ve+/ve+/ve+/ve+/ve+/ve+/vWhl77+977+977+9aQ0R77+9ClEi77+977+977+9zZMlIu+/vQZR77+977+9KO+/vQDvv73vv70aEVHvv70oHu+/ve+/vTw7IiLvv70v77+9N++/vTvvv70kIu+/vS8qTe+/vUvvv71KREQP77+9Su+/vWhdFO+/vV02Ee+/vX9BExEREREREREREe+/vVhfCSNA77+977+9Q++/vSAAAAAASUVORO+/vUJg77+9",
        "mime_type": "image/png",
      },
      Object {
        "content": "PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAyMyAyMyI+PHBhdGggZmlsbD0iYmxhY2siIGQ9Ik0xIDFoN3Y3aC03ek05IDFoNXYxaC0zdjVoLTF2LTFoLTF2LTFoMXYtMWgtMXpNMTUgMWg3djdoLTd6TTIgMnY1aDV2LTV6TTE2IDJ2NWg1di01ek0zIDNoM3YzaC0zek0xNyAzaDN2M2gtM3pNMTMgNWgxdjRoMXYxaDF2MmgtMXYtMWgtMnYtMmgtMnYxaDF2MWgtMXYxaC0xdjNoMXYxaC0xdjNoLTF2LTVoLTZ2LTFoMXYtMWgtMXYtMWgxdi0yaDZ2LTFoMXYtMWgxdjFoMXYtMWgtMXYtMWgxek05IDdoMXYxaC0xek0xIDloMXYxaC0xek0xNyA5aDF2MWgtMXpNMTkgOWgzdjNoLTF2MWgxdjFoLTF2MWgtMXYtMmgtMXYtMWgxdi0xaDF2LTFoLTF2MWgtMXpNNiAxMHYxaC0xdjJoM3YtMWgxdi0yek0xIDExaDF2MWgxdjFoLTF2MWgtMXpNNiAxMWgydjFoLTJ6TTE4IDExaDF2MWgtMXpNMTEgMTJoMnYxaC0yek0xNiAxMmgydjFoLTF2MWgtM3YtMWgyek0xOCAxM2gxdjNoMXYxaC0xdjFoLTF2LTJoLTF2LTFoMXpNMTIgMTRoMnYxaDJ2MWgtMXYxaC0xdjJoLTJ2MWgtMXYtMmgxdi0xaC0xdi0xaDJ2LTFoLTF6TTEgMTVoN3Y3aC03ek0yIDE2djVoNXYtNXpNMTYgMTZoMXYzaC0xdi0xaC0xdi0xaDF6TTIxIDE2aDF2MWgtMXpNMyAxN2gzdjNoLTN6TTE1IDE5aDF2MWgxdjFoLTN2LTFoMXpNMTcgMTloMnYxaDF2LTFoMnYyaC0zdjFoLTJ2LTFoMXYtMWgtMXpNMTAgMjBoMXYxaC0xek05IDIxaDF2MWgtMXpNMTIgMjFoMXYxaC0xeiIvPjwvc3ZnPg==",
        "mime_type": "image/svg+xml",
      },
    ],
    "status": "ACTIVE",
  },
}
`;
